test/node/storage.js

```diff
@@ -0,0 +1,20 @@
+module.exports = {
+  _store: Object.create( null ),
+  setItem: function( key, value ) {
+    this._store[ key ] = value;
+    this.length++;
+  },
+  getItem: function( key ) {
+    return key in this._store ? this._store[ key ] : null;
+  },
+  removeItem: function( key ) {
+    if ( key in this._store ) {
+      delete this._store[ key ];
+      this.length--;
+    }
+  },
+  key: function( i ) {
+    return Object.keys( this._store )[ i ];
+  },
+  length: 0
```

Comment:
Manual record-keeping would be unnecessary if this were `get length() { return Object.keys( this._store ).length; }`.